Add allergies exercise

pfertyk opened the pull request.

pfertyk commented:
@BNAndras could you please take a look? ;)

BNAndras commented:
Looks good. I’ll need to compare with the canonical data.

BNAndras commented on exercises/practice/allergies/allergies_test.gd:
I'd suggest using something to break up the test suite into more digestible chunks. The canonical data groups all the eggs tests together for example. I think the easiest would be to add a comment on a line by itself at the beginning of each section. It'd look something like this:

```
last test from previous section

# comment describing the new section

first test from the new section
```

pfertyk replied:
I just copied these tests from the Python track :) I can group them as you suggested though, no problem :+1:

pfertyk commented:
@BNAndras thank you for your review, I will update my code soon ;) Could you please check if other PRs also need any changes?

pfertyk commented:
@BNAndras Sorry for the delay :) I've added comments to split the tests into groups, like in the canonical data. I've also added a prefixe to each test to indicate the group.
